src/plots/plot.ts

```diff
@@ -80,6 +95,7 @@ export class Plot extends Component {
     super._setup();
     this._renderArea = this.content().append("g").classed("render-area", true);
     this.datasets().forEach((dataset) => this._createNodesForDataset(dataset));
+    this._createCanvas();
```

Comment:
Yeah good idea - how about this:

* Instead of having both a _renderArea SVG and a _canvas, we introduce a RenderArea class or interface which abstracts the renderArea and canvas. The RenderArea has two implementations: SVGRenderArea and CanvasRenderArea.
* Plot passes the RenderArea its _element() div and the RenderArea adds and owns the SVG/Canvas DOM elements
* We expose this API on Plot by adding `plot.renderArea("canvas") / plot.renderArea("svg")`, which we internally convert into the appropriate RenderArea class and muck the DOM appropriately
* In Plot._paint() we call appropriate methods on the RenderArea interface to draw appropriately

This way we get a separate renderer API while still keeping just one Plot hierarchy and minimal code-breakage for all our plots. Thoughts? @CalvinFernandez